File: library_net/src/main/java/com/hjc/library_net/security/AllTrustManager.java
package com.hjc.library_net.security;

import android.annotation.SuppressLint;

import java.security.cert.X509Certificate;

import javax.net.ssl.X509TrustManager;

/**
 * @Author: HJC
 * @Date: 2021/9/13 15:28
 * @Description: 证书校验(默认都是信任)
 */
@SuppressLint("CustomX509TrustManager")
public class AllTrustManager implements X509TrustManager {

    @SuppressLint("TrustAllX509TrustManager")
    @Override
    public void checkClientTrusted(X509Certificate[] chain, String authType) {

    }

    @SuppressLint("TrustAllX509TrustManager")
    @Override
    public void checkServerTrusted(X509Certificate[] chain, String authType) {

    }

    @Override
    public X509Certificate[] getAcceptedIssuers() {
        return new X509Certificate[0];
    }
}

File: library-net/src/main/java/com/hjc/library_net/security/AllTrustManager.kt
package com.hjc.library_net.security

import android.annotation.SuppressLint
import java.security.cert.X509Certificate
import javax.net.ssl.X509TrustManager

/**
 * @Author: HJC
 * @Date: 2021/9/13 15:28
 * @Description: 证书校验(默认都是信任)
 */
@SuppressLint("CustomX509TrustManager")
class AllTrustManager : X509TrustManager {

    @SuppressLint("TrustAllX509TrustManager")
    override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {

    }

    @SuppressLint("TrustAllX509TrustManager")
    override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {

    }

    override fun getAcceptedIssuers(): Array<X509Certificate> {
        return arrayOf()
    }

}